src/aics-image-viewer/components/ChannelsWidget/index.tsx

```diff
@@ -62,27 +48,32 @@ export default class ChannelsWidget extends React.Component<ChannelsWidgetProps,
     super(props);
   }
 
-  createCheckboxHandler = (key: string, value: boolean) => (channelArray: number[]) => {
+  createCheckboxHandler = (key: ChannelStateKey, value: boolean) => (channelArray: number[]) => {
     this.props.changeChannelSettings(channelArray, key, value);
   };
 
-  showVolumes = this.createCheckboxHandler(VOLUME_ENABLED, true);
-  showSurfaces = this.createCheckboxHandler(ISO_SURFACE_ENABLED, true);
-  hideVolumes = this.createCheckboxHandler(VOLUME_ENABLED, false);
-  hideSurfaces = this.createCheckboxHandler(ISO_SURFACE_ENABLED, false);
+  showVolumes = this.createCheckboxHandler("volumeEnabled", true);
+  showSurfaces = this.createCheckboxHandler("isosurfaceEnabled", true);
+  hideVolumes = this.createCheckboxHandler("volumeEnabled", false);
+  hideSurfaces = this.createCheckboxHandler("isosurfaceEnabled", false);
 
   renderVisibilityControls(channelArray: number[]) {
     const { channelSettings, channelDataChannels } = this.props;
 
-    const arrayOfNames = channelArray.map((channelIndex: number) => channelDataChannels[channelIndex].name);
-    const volChecked = arrayOfNames.filter((name: string) => {
+    let volChecked: number[] = [];
+    let isoChecked: number[] = [];
+    channelArray.forEach((channelIndex: number) => {
+      const name = channelDataChannels![channelIndex].name;
       const channelSetting = find(channelSettings, { name });
-      return channelSetting && channelSetting[VOLUME_ENABLED];
-    });
-    const isoChecked = arrayOfNames.filter((name: string) => {
-      const channelSetting = find(channelSettings, { name });
-      return channelSetting && channelSetting[ISO_SURFACE_ENABLED];
+      if (!channelSetting) return;
+      if (channelSetting.volumeEnabled) {
+        volChecked.push(channelIndex);
+      }
+      if (channelSetting.isosurfaceEnabled) {
+        isoChecked.push(channelIndex);
+      }
```

Comment:
why did this function implementation change?

Reply:
`SharedCheckbox` expects two lists of keys: `allOptions` for the options it controls, and `checkedList` for the options which are currently checked. TypeScript revealed that these lists were different types: `allOptions` was a list of channel indexes, while `checkedList` was a list of channel names. Bizarrely, this worked just fine before, as only the length and not the content of `checkedList` ever matters for the component's behavior. Regardless, it was still a type error (and this list was still being built by two loops when one would do).